config/initializers/session_store.rb

```diff
@@ -32,6 +32,23 @@
       :expire_after => 24.hours,
       :key          => "_vmdb_session",
     )
+
+    require 'rack/session/dalli'
+    Rack::Session::Dalli.prepend Module.new {
+      # As we monkey-patch marshal to support autoloading, Dalli can
+      # cause a load to occur. Consequently, we need to manage things
+      # carefully to prevent a deadlock between the Rails Interlock and
+      # Dalli's own exclusive lock.
+      def with_lock(*args)
+        ActiveSupport::Dependencies.interlock.permit_concurrent_loads do
```

Comment:
@matthewd is there a way to test this in case the interface of dalli or the interlock code changes?